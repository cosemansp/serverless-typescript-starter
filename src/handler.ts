import { successResponse, runWarm } from './core';
import { Context, Callback, Handler } from 'aws-lambda';

const handler: Handler = async (event: any, context: Context, callback: any): Promise<any> => {
  const response = successResponse({
    message: 'Go Serverless v1.1! Your function executed successfully!',
    input: event,
  });
  callback(null, response);
};

// runWarm function handles pings from the scheduler so you don't
// have to put that boilerplate in your function.
export default runWarm(handler);
