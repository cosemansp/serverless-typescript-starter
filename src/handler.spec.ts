import handler from './handler';

describe('handler', () => {
  it('executes as expected', async () => {
    const result = await handler({}, {} as AWSLambda.Context);
    expect(result).toMatchSnapshot();
  });
});
