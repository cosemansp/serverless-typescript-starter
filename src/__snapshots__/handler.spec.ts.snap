// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handler executes as expected 1`] = `
Object {
  "body": "{\\"message\\":\\"Go Serverless v1.0! Your function executed successfully!\\",\\"input\\":{}}",
  "headers": Object {},
  "statusCode": 200,
}
`;
